src/Sentry/Scope.cs

```diff
@@ -127,13 +127,16 @@ public User User
         };
         set
         {
-            _user = value;
-            if (_user is not null)
+            if (_user != value)
```

Comment:
lets keep it consistent (using pattern matching below and operator here):

```suggestion
            if (_user is not value)
```